src/backend/common/DependencyModule.cpp

```diff
@@ -19,30 +20,74 @@
 #include <dlfcn.h>
 #endif
 
+using common::Version;
+using std::make_tuple;
+using std::string;
+using std::to_string;
+using std::vector;
+
+constexpr Version NullVersion{0, 0, 0};
+
 #ifdef OS_WIN
 #include <Windows.h>
+
 static const char* librarySuffix = ".dll";
-static const char* libraryPrefix = "";
+
+namespace {
+vector<string> libName(const std::string& name, const string& suffix,
+                       const Version& ver = NullVersion) {
+    UNUSED(ver);  // Windows DLL files are not version suffixed
+    return {name + suffix + librarySuffix};
+}
+}  // namespace
+
 #elif defined(OS_MAC)
+
 static const char* librarySuffix = ".dylib";
 static const char* libraryPrefix = "lib";
+
+namespace {
+vector<string> libName(const std::string& name, const string& suffix,
+                       const Version& ver = NullVersion) {
+    UNUSED(suffix);
+    if (ver != NullVersion) {
+        const string infix = "." + to_string(std::get<0>(ver)) + ".";
+        return {libraryPrefix + name + infix + librarySuffix,
+                libraryPrefix + name + librarySuffix};
+    } else {
+        return {libraryPrefix + name + librarySuffix};
+    }
+}
+}  // namespace
+
 #elif defined(OS_LNX)
+
 static const char* librarySuffix = ".so";
 static const char* libraryPrefix = "lib";
-#else
-#error "Unsupported platform"
-#endif
-
-using std::string;
-using std::vector;
 
 namespace {
+vector<string> libName(const std::string& name, const string& suffix,
+                       const Version& ver = NullVersion) {
+    UNUSED(suffix);
+    if (ver != NullVersion) {
+        const string soname("." + to_string(std::get<0>(ver)));
 
-std::string libName(const std::string& name) {
-    return libraryPrefix + name + librarySuffix;
+        const string vsfx = "." + to_string(std::get<0>(ver)) + "." +
+                            to_string(std::get<1>(ver)) + "." +
+                            to_string(std::get<2>(ver));
+        return {libraryPrefix + name + librarySuffix + vsfx,
```

Comment:
libraryPrefix + name+ librarySuffix  is used multiple times. You should store that in a temporary and add the values that are different for the values. This is will most likely require multiple allocations.

Reply:
:+1: took care